haystack_experimental/components/embedders/sentence_transformers_doc_image_embedder.py

```diff
@@ -0,0 +1,335 @@
+                images_paths_types.append({"path": resolved_file_path, "type": "pdf", "page_number": page_number})
+            else:
+                images_paths_types.append({"path": resolved_file_path, "type": "image"})
+
+        return images_paths_types
+
+    @staticmethod
+    def _process_pdf_documents(
+        images_to_embed: List,
+        pdf_documents: List[Dict[str, Any]],
+        size: Optional[Tuple[int, int]],
+    ) -> None:
+        """
+        Process PDF documents and populate the images_to_embed list with converted images.
+
+        :param images_to_embed: List to populate with converted PIL images (modified in place).
+        :param pdf_documents: List of dictionaries with doc_idx, path, and page_number.
+        :param size: Optional tuple of width and height to resize the images to.
+        """
+        if not pdf_documents:
+            return
+
+        pdf_files_by_path = defaultdict(list)
+        for pdf_doc in pdf_documents:
+            pdf_files_by_path[pdf_doc["path"]].append((pdf_doc["doc_idx"], pdf_doc["page_number"]))
+
+        # Open and convert each PDF file once
+        for file_path, doc_page_pairs in pdf_files_by_path.items():
+            page_numbers = [page_num for _, page_num in doc_page_pairs]
+            bytestream = ByteStream.from_file_path(file_path)
+            pdf_images = _convert_pdf_to_pil_images(bytestream=bytestream, page_range=page_numbers, size=size)
+            # Map back to document positions
+            page_to_pil_image = dict(pdf_images)
+            for doc_idx, page_num in doc_page_pairs:
+                images_to_embed[doc_idx] = page_to_pil_image[page_num]
+
+    @component.output_types(documents=List[Document])
+    def run(self, documents: List[Document]) -> Dict[str, List[Document]]:
+        """
+        Embed a list of documents.
+
+        :param documents:
+            Documents to embed.
+
+        :returns:
+            A dictionary with the following keys:
+            - `documents`: Documents with embeddings.
+        """
+        if not isinstance(documents, list) or documents and not isinstance(documents[0], Document):
+            raise TypeError(
+                "SentenceTransformersDocumentImageEmbedder expects a list of Documents as input. "
+                "In case you want to embed a list of strings, please use the SentenceTransformersTextEmbedder."
+            )
+        if self._embedding_backend is None:
+            raise RuntimeError("The embedding model has not been loaded. Please call warm_up() before running.")
+
+        images_paths_types = self._validate_image_paths(documents)
+
+        images_to_embed: List = [None] * len(documents)
+        pdf_documents = []
+
+        for doc_idx, image_path_type in enumerate(images_paths_types):
+            if image_path_type["type"] == "image":
+                # Process images directly
+                image: Union["Image", "ImageFile"] = PILImage.open(image_path_type["path"])
```

Comment:
Part of this [comment](https://github.com/deepset-ai/haystack-experimental/pull/319/files#r2137082791). I think we should keep using `IMAGE_MIME_TYPES` as the way to determine if a file is an "image" or not.